polaris-react/src/components/TextField/TextField.scss

```diff
@@ -23,13 +23,15 @@ $spinner-icon-size: 12px;
   // only show the clear button when focused within the textfield
   &:focus-within {
     .ClearButton {
+      display: block;
       visibility: visible;
       opacity: 1;
     }
   }
 
   &:not(:focus-within) {
     .ClearButton {
+      display: none;
```

Comment:
This solves the bug where the clear button is rendering when invisible

|Before|After|
|-|-|
|![Screen Recording 2023-11-21 at 5 04 49 PM](https://github.com/Shopify/polaris/assets/20652326/8b2406af-e18c-4c68-82a8-0722d23acd6a)|![Screen Recording 2023-11-21 at 5 03 34 PM](https://github.com/Shopify/polaris/assets/20652326/65de46a4-8a7b-4e22-85b4-aecf700dfd4a)|